toolkitcore/Controllers/ChatCommandController.cs

```diff
@@ -12,7 +12,172 @@ public static class ChatCommandController
     {
         public static ToolkitChatCommand GetChatCommand(string commandText)
         {
-            return DefDatabase<ToolkitChatCommand>.AllDefs.ToList().Find(cc => string.Equals(cc.commandText, commandText, StringComparison.InvariantCultureIgnoreCase));
+            var baseCommand = Parse(commandText).FirstOrDefault();
+
+            if (baseCommand == null)
+            {
+                return null;
+            }
+
+            return DefDatabase<ToolkitChatCommand>.AllDefsListForReading.FirstOrDefault(
+                c => c.commandText.EqualsIgnoreCase(baseCommand)
+            );
+        }
+
+        public static IEnumerable<string> Parse(string input, string prefix = "!")
+        {
+            // Strip the prefix from the input (if it exists)
+            if (input.ToLowerInvariant().StartsWith(prefix))
```

Comment:
Command text input will never have '!'. ChatMessages without a '!' are never parsed to begin with.